### PR TITLE
fix: fix failing tests for 'annotation'

### PR DESCRIPTION
**Pull request by spiridonovfed**

### CODE REVIEW

File: annotation/tests/test_category_crud.py [hunk 1]
```diff
@@ -14,6 +15,23 @@
 
 client = TestClient(app)
 
+ATTRIBUTES_NOT_IN_CATEGORY_MODEL = ["is_leaf"]
+NOT_FULLY_TEST_SUPPORTED_CATEGORY_ATTRIBUTES = [
+    "children",
+    "parents",
+    "is_leaf",
+    "id",
+]
```
Comment (cakeinsauce) [inline annotation/tests/test_category_crud.py:24]: Just a kind reminder about `tuple` =)

File: annotation/app/categories/services.py [hunk 2]
```diff
@@ -73,7 +73,7 @@ def add_category_db(
         tree = Ltree(f"{id_}")
 
     category = Category(
-        id=(id_ or str(uuid.uuid4())),
+        id=id_,
```
Comment (spiridonovfed) [inline annotation/app/categories/services.py:76]: change suggested by Kirill - developer of this functionality

File: annotation/app/categories/services.py [hunk 3]
```diff
@@ -278,8 +272,10 @@ def _get_parents(
     uniq_pathes = set()
 
     for cat in categories:
-        uniq_pathes.add(cat.tree.path)
-        uniq_cats = uniq_cats.union({tree.path for tree in cat.tree})
+        # if we pass root categories it causes exception.
+        if cat.tree is not None:
+            uniq_pathes.add(cat.tree.path)
+            uniq_cats = uniq_cats.union({tree.path for tree in cat.tree})
```
Comment (spiridonovfed) [inline annotation/app/categories/services.py:279]: change suggested by Kirill - developer of this functionality

**Review by cakeinsauce — CHANGES_REQUESTED** [review]

File: annotation/tests/test_category_crud.py (hunk 1 above)
Comment (spiridonovfed) [reply annotation/tests/test_category_crud.py]: agreed - changed to tuples

File: annotation/app/schemas/categories.py [hunk 4]
```diff
@@ -62,6 +62,7 @@ class Config:
 
 class CategoryResponseSchema(CategoryInputSchema):
     parents: Optional[List[dict]] = Field()
+    children: Optional[List[dict]] = Field()
```
Comment (MagicTearsAsunder) [inline annotation/app/schemas/categories.py]: У нас только parents и is_leaf,
children - легаси. 
нужно поправить тесты так, чтобы они не использовали children

File: annotation/app/categories/services.py [hunk 5]
```diff
@@ -33,19 +33,43 @@
...
+    parents: Optional[List[CategoryResponseSchema]] = None,
+    is_leaf: bool = False,
+) -> CategoryResponseSchema:
+    if parents is None:
+        parents = []
+    category_response = response_object_from_db(category_db)
+    category_response.is_leaf = is_leaf
+    category_response.parents = parents
+    return category_response
+
```
Comment (spiridonovfed) [inline annotation/app/categories/services.py:60]: change suggested by Kirill

**Review by MagicTearsAsunder — APPROVED** [review]

File: annotation/app/schemas/categories.py (hunk 4 above)
Comment (spiridonovfed) [reply annotation/app/schemas/categories.py]: Agreed - fixed

**Pull request merged**